bundles/binding/org.openhab.binding.lightwaverf/src/main/java/org/openhab/binding/lightwaverf/internal/LightwaveRFReceiver.java

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.lightwaverf.internal;
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.Enumeration;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRFCommand;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfCommandOk;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfHeatInfoRequest;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfRoomDeviceMessage;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfRoomMessage;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfSerialMessage;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfVersionMessage;
+import org.openhab.binding.lightwaverf.internal.exception.LightwaveRfMessageException;
+import org.openhab.binding.lightwaverf.internal.message.LightwaveRFMessageListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+  * @author Neil Renaud
+  * @since 1.7.0
+  */
+public class LightwaveRFReceiver implements Runnable {
+    private static final Logger logger = LoggerFactory.getLogger(LightwaveRFReceiver.class);
+    private static final int DELAY_BETWEEN_RECEIVES_MS = 10;
+    
+    private final CopyOnWriteArrayList<LightwaveRFMessageListener> listeners = new CopyOnWriteArrayList<LightwaveRFMessageListener>();
+    private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
```

Comment:
It's a scheduled thread pool so allows us to do "scheduleWithFixedDelay" instead of creating our own thread with a sleep which (in my view is more messy). Also means we don't have to worry about cancelling threads or threads dying as the Threadpool will handle it.